src/main/java/com/minecolonies/coremod/colony/buildings/inventory/ItemSearchResult.java

```diff
@@ -0,0 +1,67 @@
+package com.minecolonies.coremod.colony.buildings.inventory;
+
+import com.minecolonies.api.crafting.ItemStorage;
+import com.minecolonies.coremod.tileentities.TileEntityRack;
+import net.minecraft.item.ItemStack;
+
+import java.util.Set;
+
+/**
+ * Wraps all needed information returned by an item search in a class.
+ */
+public class ItemSearchResult
```

Comment:
Again this is completely locked to Racks and not what is actually wanted or needed by our players.
ItemSearch should happen colony wide if possible not limited to racks or a specific inventory type.